drivers/mfd/tssupervisor-core.c

```diff
@@ -0,0 +1,246 @@
+			      const char *buf, size_t count)
+{
+	struct ts_supervisor *super = dev_get_drvdata(dev);
+	unsigned int ctrl_reg = 0;
+	bool en;
+	int ret;
+
+	ret = kstrtobool(buf, &en);
+	if (ret)
+		return ret;
+
+	if (en)
+		ctrl_reg |= FLG_WAKE_EN;
+
+	ret = regmap_update_bits(super->regmap, SUPER_FLAGS,
+				  FLG_WAKE_EN,
+				  ctrl_reg);
+
+	return ret ? ret : count;
+}
+
+static ssize_t wake_en_show(struct device *dev,
+				struct device_attribute *attr, char *buf)
+{
+	struct ts_supervisor *super = dev_get_drvdata(dev);
+	unsigned int reg;
+	int ret;
+
+	ret = regmap_read(super->regmap, SUPER_FLAGS, &reg);
+	if (ret)
+		return ret;
+	ret = sprintf(buf, "%d\n", !!(reg & FLG_WAKE_EN));
+	return ret;
+}
+static DEVICE_ATTR_RW(wake_en);
+
+static ssize_t console_cfg_store(struct device *dev, struct device_attribute *attr,
+			      const char *buf, size_t count)
+{
+	struct ts_supervisor *super = dev_get_drvdata(dev);
+	unsigned int ctrl_reg;
+	int ret;
+
+	if (sysfs_streq(buf, "auto"))
+		ctrl_reg = 0;
+	else if (sysfs_streq(buf, "always-usb"))
+		ctrl_reg = FLG_FORCE_USB_CON;
+	else
+		return -EINVAL;
+
+	ret = regmap_update_bits(super->regmap, SUPER_FLAGS, FLG_FORCE_USB_CON,
+				 ctrl_reg);
+
+	return ret ? ret : count;
+}
+
+static ssize_t console_cfg_show(struct device *dev,
+				struct device_attribute *attr, char *buf)
+{
+	struct ts_supervisor *super = dev_get_drvdata(dev);
+	unsigned int reg;
+	int ret;
+
+	ret = regmap_read(super->regmap, SUPER_FLAGS, &reg);
+	if (ret)
+		return ret;
+	
+	if (reg & FLG_FORCE_USB_CON)
+		ret = sprintf(buf, "auto [always-usb]\n");
+	else
+		ret = sprintf(buf, "[auto] always-usb\n");
+
+	return ret;
+}
+static DEVICE_ATTR_RW(console_cfg);
+
+static struct attribute *ts7250v3_sysfs_entries[] = {
+	&dev_attr_vbus_present.attr,
+	&dev_attr_wake_en.attr,
+	&dev_attr_console_cfg.attr,
+	NULL,
+};
+
+static struct attribute_group ts7250v3_attr_group = {
+	.attrs	= ts7250v3_sysfs_entries,
+};
+
+static int ts_supervisor_i2c_probe(struct i2c_client *client)
+{
+	struct ts_supervisor *super;
+	struct device *dev = &client->dev;
+	int err = 0, i;
+	uint32_t model, revision;
+
+	super = devm_kzalloc(dev, sizeof(struct ts_supervisor),
+			     GFP_KERNEL);
+	if (!super)
+		return -ENOMEM;
+
+	dev_set_drvdata(dev, super);
+
+	super->client = client;
+	super->regmap = devm_regmap_init_i2c(client, &ts_supervisor_i2c_regmap);
+	if (IS_ERR(super->regmap)) {
+		err = PTR_ERR(super->regmap);
+		dev_err(dev, "Failed to allocate register map: %d\n", err);
+		return err;
+	}
+
+	err = regmap_read(super->regmap, SUPER_MODEL, &model);
+	if (err < 0)
+		dev_err(dev, "error reading reg %u", SUPER_MODEL);
+	err = regmap_read(super->regmap, SUPER_REV_INFO, &revision);
+	if (err < 0)
+		dev_err(dev, "error reading reg %u", SUPER_REV_INFO);
+	dev_info(&client->dev, "Model %04X rev %d%s\n",
+		 model,
+		 revision & 0x7fff,
+		 revision & 0x8000 ? " (DIRTY)" : "");
+
+	if (model == MODEL_TS_7250_V3) {
```

Comment:
What is the reason for locking this driver to the 7250v3? (or at least its creation of sysfs entries ... otherwise succeeding but not creating them?). We might want a more maintainable (and generic) way to make these determinations while at the same time supporting a lot of new models that won't ID as 7250.